File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/minimum-number-of-operations-to-make-elements-in-array-distinct.cpp
class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        int count=0;
        set<int>unq(nums.begin(),nums.end());
        if(unq.size()==nums.size()){
            return 0;
        }
        while(unq.size()!=nums.size()&&nums.size()>3){
            nums.erase(nums.begin(),nums.begin()+3);
            count++;
            unq.clear();
            unq.insert(nums.begin(),nums.end());
        }
        if (nums.size()==unq.size()){
            return count;
        }
        if(nums.size()!=unq.size() && nums.size()<=3){
            count++;
        }
        return count;

    }
};

File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/minimum-number-of-operations-to-make-elements-in-array-distinct.py
class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        count=0
        if len(nums)==len(set(nums)):
            return 0
        while len(nums)!=len(set(nums)) and len(nums)>3:
            del nums[:3]
            count+=1
        if len(nums)==len(set(nums)):
            return count 
        if len(nums)!=len(set(nums)) and len(nums)<=3:
            count+=1
        return count
        

        